flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateFactory.java

```diff
@@ -121,8 +127,8 @@ else if (partitionLocation.isRemote()) {
 					connectionManager,
 					partitionRequestInitialBackoff,
 					partitionRequestMaxBackoff,
-					metrics
-				);
+					metrics,
+					memorySegmentProvider);
```

Comment:
Whether to separate line for last `);` which keeps the consistent with `local` and `unknown` cases?